core/src/components/modal/gestures/swipe-to-close.ts

```diff
@@ -24,17 +26,34 @@ export const createSwipeToCloseGesture = (el: HTMLIonModalElement, animation: An
       return true;
     }
 
-    const contentOrFooter = target.closest('ion-content, ion-footer');
-    if (contentOrFooter === null) {
+    /**
+     * If we are swiping on the content,
+     * swiping should only be possible if
+     * the content is scrolled all the way
+     * to the top so that we do not interfere
+     * with scrolling.
+     */
+    const content = target.closest('ion-content');
+    if (content) {
+      const root = getElementRoot(content);
+      const scrollEl = root.querySelector('.inner-scroll') as HTMLElement;
```

Comment:
Are there any considerations here for `.ion-content-scroll-host` and virtual scroll implementations?

Reply:
👍 good catch will fix